File: excyrender/Photometry/SPD/Constant.hs
-- (C) 2013 Sebastian Mach (1983), this file is published under the terms of the
-- GNU General Public License, Version 3 (a.k.a. GPLv3).
-- See COPYING in the root-folder of the excygen project folder.

module Photometry.SPD.Constant
( constantSPD
) where

import Photometry.SPD.SPD
import Photometry.SPD.Regular (regularSPD)
import Photometry.CIEMatchingCurves (cie_start, cie_end)


-- Constant ------------------------------------------------------------------------------------
constantSPD :: (RealFrac a) => a -> SPD a

constantSPD x = SPD {
                    sample = \_ -> x,
                    toXYZ  = toXYZ (regularSPD cie_start cie_end [x]),
                    stretch = \f -> constantSPD (f*x)
                }

File: excyrender.cxx/Photometry/SPD/Constant.cc
// (C) 2013 Sebastian Mach (1983), this file is published under the terms of the
// GNU General Public License, Version 3 (a.k.a. GPLv3).
// See COPYING in the root-folder of the excygen project folder.

#include "Constant.hh"
#include "Photometry/SPD/Regular.hh"
#include "Photometry/CIEMatchingCurves.hh"
#include <array>

namespace excyrender { namespace Photometry { namespace SPD {

    std::tuple<real,real,real> Constant::toXYZ() const
    {
        using namespace CIEMatchingCurves;
        return Regular(cie_start, cie_end, std::array<real,1>{{c_}}).toXYZ();
    }

} } }
